x/mint/keeper/keeper_test.go

```diff
@@ -719,3 +719,348 @@ func (suite *KeeperTestSuite) TestDistributeDeveloperRewards() {
 		})
 	}
 }
+
+// TestCalculateTruncationsDelta tests that the per-epoch truncation delta is calculated
+// correctly. That includes checking that the pre-existing delta is
+// extracted from the store at the desired index and summed with the new delta
+// from the input values.
+func (suite *KeeperTestSuite) TestCalculateTruncationDelta() {
+	const mintDenom = sdk.DefaultBondDenom
+	tests := map[string]struct {
+		preExistingStoreDelta sdk.Dec
+
+		moduleAccountName string
+		provisions        sdk.Dec
+		amountDistributed sdk.Int
+
+		expectedTruncationDelta sdk.Dec
+
+		expectError bool
+	}{
+		"100.6 - 100 = 0.6; no pre-existing delta; inflation delta module account": {
+			moduleAccountName: types.DeveloperVestingModuleAcctName,
+			provisions:        sdk.NewDecWithPrec(1006, 1),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedTruncationDelta: sdk.NewDecWithPrec(6, 1),
+		},
+		"100.6 - 100  + 0.3 = 0.9; pre-existing delta of 0.3; inflation delta module account": {
+			preExistingStoreDelta: sdk.NewDecWithPrec(3, 1),
+
+			moduleAccountName: types.ModuleName,
+			provisions:        sdk.NewDecWithPrec(1006, 1),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedTruncationDelta: sdk.NewDecWithPrec(9, 1),
+		},
+		"100.6 - 100  + 0.5 = 1.1; pre-existing delta of 1.1; inflation delta module account": {
+			preExistingStoreDelta: sdk.NewDecWithPrec(5, 1),
+
+			moduleAccountName: types.ModuleName,
+			provisions:        sdk.NewDecWithPrec(1006, 1),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedTruncationDelta: sdk.NewDecWithPrec(11, 1),
+		},
+		"82191.178 - 17808  + 33.33 = 64416508; pre-existing delta of 33.33; developer vesting module account": {
```

Comment:
I though the delta shouldn't exceed 1?

Reply:
The goal is to test all possible combinations a value can take, not only what we expect

Reply:
Should we make a comment somewhere explaining that this state is not expected to be reached however we are still testing for it? I only say this because it threw me off when I was reviewing.

Side note: Is there a reason we don't force the value to be less than 1 (i.e. if the delta were to be returned as anything less than 1 then we panic?)

Reply:
Technically, it is still allowed to have > 1 truncation delta. For example, we might want to manually set it in the upgrade handler to be auto-distributed at the next epoch. On the other hand, we can also directly distribute it to the community pool instead too.

I will think about this suggestion.

Reply:
I think a guarantee of having a truncation delta less than 1 is the safer / more preferred method IMO. Especially since we are minting whatever this value is. Lmk what you think

Reply:
Added a check per suggestion